src/screens/VaultScreen.js

```diff
@@ -13,11 +14,305 @@ export default class VaultScreen extends React.Component {
+    for(var i = 0; i < resources_arr.length; i++) {
+      if(resources_arr[i].id == item_id) {
+        resources_arr[i].upvotes = resources_arr[i].upvotes+1;
+      }
+    };
+    this.setState({resources:resources_arr});
+  }
+  displayResources() {
+    return this.state.resources.map((item) => {
+      return (
+        <View key = {item.id} style={styles.contentPanel}>
+          <Text style={styles.contentTitle}>{item.title}</Text>
+          <View style={styles.contentInformation}>
+            <View style={{justifyContent:'center'}}>
+              <Text style={styles.partnerOrg}>{item.partner_org}</Text>
+            </View>
+            <View style={{justifyContent:'center', marginLeft: 5,}}>
+              <Text style={styles.dateText}>{item.date}</Text>
+            </View>
+          </View>
+          <Text style={[styles.bodyText, {marginTop: 10,}]}>{item.description}</Text>
+          <View style={[styles.contentInformation, {marginTop: 10,}]}>
+            <View style={styles.button}>
+              <Text style={{color:'white', fontSize:12, fontFamily: 'source-sans-pro-semibold',}}>OPEN RESOURCE</Text>
+            </View>
+            <View style={styles.upvote}>
+              <TouchableHighlight onPress={() => {this.upvote(item.id);}}>
+                <Text style={styles.upvoteText}>{item.upvotes}</Text>
+              </TouchableHighlight>
+            </View>
+          </View>
+        </View>
+      );
+    })
+  }
+  onSubmitEdit = () => {
+    //add options
+  }
   render() {
     return (
-      <View style={layoutStyles.flexCenter}>
-        <Text>This is the Vault screen!</Text>
-      </View>
+      <ScrollView contentContainerStyle={{flexGrow:1}} >
+        {
+          this.state.fontLoaded? (
+            <View style={{flex:1}}>
+              <View style={styles.backgroundContainer}>
+                <View style={styles.backgroundDisplay}>
+                </View>
+                <View style={[styles.contentContainer, {top:0,}]}>
+                  <Text style={[styles.baseText, {marginBottom: 5,}]}>
+                    <Text style={[styles.titleText, {color:'rgb(255, 255, 255)'}]}>
+                      {this.state.titleName}
+                    </Text>
+                  </Text>
+                  <View style={[styles.search, {marginTop: 5,}]}>
+                    <TextInput style={styles.searchBar} placeholderTextColor="rgba(255, 255, 255, 0.5)" placeholder="Search resources" onChangeText={(searchText) => this.setState({searchText})}/>
+                  </View>
+                </View>
+                <ScrollView horizontal={true} showsHorizontalScrollIndicator={false} style={styles.filter}>  
+                  {this.filterScroller()}
+                </ScrollView>
+                <View style={[styles.contentContainer, {top:125,}]}>
+                  {this.displayResources()}
+                </View>
+              </View>
+            </View>
+          ) : null
```

Comment:
I like the way that you handled the conditional here but I would probably abstract this logic into a helper function that renders the ScrollView so the code here isn't too deeply nested.